components/playlist/media_detector_component_manager.cc

```diff
@@ -260,14 +108,27 @@ bool MediaDetectorComponentManager::ShouldHideMediaSrcAPI(
                               });
 }
 
-const std::string& MediaDetectorComponentManager::GetMediaDetectorScript() {
-  if (!script_.empty())
-    return script_;
+std::string MediaDetectorComponentManager::GetMediaDetectorScript(
+    const GURL& url) {
+  std::string detector_script = script_;
+  if (detector_script.empty()) {
+    // In case we have yet to fetch the script, use local script instead. At the
+    // same time, fetch the script from component.
+    RegisterIfNeeded();
+    detector_script = GetLocalScript();
+  }
+
+  if (net::SchemefulSite site(url); site_specific_detectors_.count(site)) {
```

Comment:
2. And then we're going to combine that per-site script with the base script only when the site matches here.